src/view/Package.ts

```diff
@@ -0,0 +1,145 @@
+import { Option } from "../utils/monads";
+import { Widget } from "./Widget";
+import { AnyCharacterOf } from "./widgets/types/StringTypes";
+import { Template } from "./widgets/types/Template";
+
+export type WidgetClass = abstract new (...args: any[]) => Widget;
+
+export interface Prototype extends Object {
+    constructor: WidgetClass;
+}
+
+export class Resources {
+
+    public static fromCSS(css: string): Resources {
+        return new Resources(css, undefined);
+    }
+
+    public static fromJS(js: string): Resources {
+        return new Resources(undefined, js);
+    }
+
+    public static fromResources(resources: {
+        css: string,
+        js: string
+    }): Resources {
+        return new Resources(resources.css, resources.js);
+    }
+
+    public static fromJSTemplate<T extends string>(template: T, ...args: Template<T>): Resources {
+        return new Resources(undefined, template.replace(/%([ojdfislba])/g, (_, type: AnyCharacterOf<"ojdfislba">) => {
```

Comment:
Based on some confusion in index-chat, I'm parsing the format `%{letter which represents a type}`. The Template type converts this into a tuple of arg types and the code below will serialize it with JSON unless it's a literal string which shouldn't get the quotes JSON.stringify provides.